diff --git a/src/CommandLine/Core/SpecificationPropertyRules.cs b/src/CommandLine/Core/SpecificationPropertyRules.cs
index d44b29c..31a2002 100644
--- a/src/CommandLine/Core/SpecificationPropertyRules.cs
+++ b/src/CommandLine/Core/SpecificationPropertyRules.cs
@@ -103,0 +104 @@ namespace CommandLine.Core
+                                           where o.Group.IsNothing()
@@ -115,0 +117 @@ namespace CommandLine.Core
+                            where o.Group.IsNothing()
diff --git a/src/CommandLine/OptionAttribute.cs b/src/CommandLine/OptionAttribute.cs
index ce2e1fa..4a38a4d 100644
--- a/src/CommandLine/OptionAttribute.cs
+++ b/src/CommandLine/OptionAttribute.cs
@@ -110 +110 @@ namespace CommandLine
-        /// Gets or sets the option group name. When one or more options are grouped, at least one of them should have value.
+        /// Gets or sets the option group name. When one or more options are grouped, at least one of them should have value. Required rules are ignored.
